### PR TITLE
Add taskflow

### PR DESCRIPTION
**Pull request by mdcoury**

### CODE REVIEW

**Comment by mdcoury** [comment]
"make all" seems to succeed about 75% of the time, but the rest causes a seg-fault

File: .circleci/config.yml [hunk 1]
```diff
@@ -3,36 +3,11 @@ version: 2.1
 orbs:
   win: circleci/windows@2.2.0
 
-commands:
```
Comment (arcondello) [inline .circleci/config.yml:6]: looks like something went wrong on the rebase. This is all reverting changes we made since https://github.com/dwavesystems/dwave-preprocessing/pull/55 was made.

The only changes to CI should be adding 
```yml
      - run: git submodule sync
      - run: git submodule update --init
```
to most of the jobs

File: dwave/preprocessing/include/dwave-preprocessing/implication_network.hpp [hunk 2]
```diff
@@ -116,7 +117,7 @@ class stronglyConnectedComponentsInfo {
   stronglyConnectedComponentsInfo(int num_components,
                                   std::vector<int> vertex_to_component_map,
                                   mapper_t &mapper)
-      : num_components(num_components), num_vertices(mapper.num_vertices()),
+      : num_vertices(mapper.num_vertices()), num_components(num_components),
```
Comment (arcondello) [inline dwave/preprocessing/include/dwave-preprocessing/implication_network.hpp:120]: I am fine with cleaning up some of the warnings, though a bigger refactor of the roof duality stuff is needed. But ideally that would be in a separate PR.

File: dwave/preprocessing/include/dwave/presolve.hpp [hunk 3]
```diff
@@ -182,8 +194,34 @@ class Presolver {
         }
     }
 
-    // todo: break into separate presolver
-    void substitute_self_loops() {
+    //----- One-time Techniques -----//
+
+    void technq_spin_to_binary() {
```
Comment (arcondello) [inline dwave/preprocessing/include/dwave/presolve.hpp:199]: IMO if we want to prepend all of these with "technique", we might as well prepend the entire word. Only two more characters

File: dwave/preprocessing/include/dwave/presolve.hpp [hunk 4]
```diff
@@ -148,6 +156,10 @@ class Presolver {
     const Postsolver<bias_type, index_type, assignment_type>& postsolver() const;
 
  private:
+    tf::Taskflow taskflowOneTime_;
```
Comment (arcondello) [inline dwave/preprocessing/include/dwave/presolve.hpp:159]: why break these into three rather than one big flow?

File: dwave/preprocessing/include/dwave/presolve.hpp [hunk 5]
```diff
@@ -138,6 +141,11 @@ class Presolver {
     /// This clears the model from the presolver.
     model_type detach_model();
 
+    void load_taskflow_one_time();
```
Comment (arcondello) [inline dwave/preprocessing/include/dwave/presolve.hpp:144]: Not sure if this is intermediate or final design, I think ultimately we want `apply()` to just run the taskflow, whatever is there. And then the `load_` methods will determine what is in that flow.

File: dwave/preprocessing/include/dwave-preprocessing/implication_network.hpp (hunk 2 above)
Comment (mdcoury) [reply dwave/preprocessing/include/dwave-preprocessing/implication_network.hpp:120]: Yes.  I gave up on fixing the warnings: there were just too many.  Instead, I suppressed the most egregious offenders to mitigate some of the noise

File: dwave/preprocessing/include/dwave/presolve.hpp (hunk 4 above)
Comment (mdcoury) [reply dwave/preprocessing/include/dwave/presolve.hpp:159]: That's mostly for a logical separation of the 3 different "types" of techniques. I'll do some more reading on the pros/cons and leave a note in the code either way.

**Comment by mdcoury** [comment]
Splitting changes into distinct PRs
